conformance/tests/gateway-static-addresses.go

```diff
@@ -0,0 +1,118 @@
+		currentGW := &v1beta1.Gateway{}
+		err := s.Client.Get(ctx, gwNN, currentGW)
+		require.NoError(t, err, "error getting Gateway: %v", err)
+		require.Len(t, currentGW.Spec.Addresses, 3, "expected 3 addresses on the Gateway, one invalid, one usable and one unusable. somehow got %d", len(currentGW.Spec.Addresses))
+		invalidAddress := currentGW.Spec.Addresses[0]
+		unusableAddress := currentGW.Spec.Addresses[1]
+		usableAddress := currentGW.Spec.Addresses[2]
+
+		t.Logf("verifying that the Gateway %s/%s is NOT accepted due to an address type that the implementation doesn't support", gwNN.Namespace, gwNN.Name)
+		kubernetes.GatewayAcceptedConditionMustEqual(t, currentGW, metav1.ConditionFalse, v1beta1.GatewayReasonUnsupportedAddress)
+
+		t.Logf("patching Gateway %s/%s to remove the invalid address", gwNN.Namespace, gwNN.Name)
+		updatedGW := currentGW.DeepCopy()
+		updatedGW.Spec.Addresses = filterAddr(currentGW.Spec.Addresses, invalidAddress)
+		err = s.Client.Patch(ctx, updatedGW, client.MergeFrom(currentGW))
+		require.NoError(t, err, "failed to patch Gateway: %v", err)
+		kubernetes.NamespacesMustBeReady(t, s.Client, s.TimeoutConfig, namespaces)
+		kubernetes.GatewayMustHaveLatestConditions(t, s.Client, s.TimeoutConfig, gwNN)
+
+		t.Logf("verifying that the Gateway %s/%s is now accepted, but is not programmed due to an address that can't be used", gwNN.Namespace, gwNN.Name)
+		err = s.Client.Get(ctx, gwNN, currentGW)
+		require.NoError(t, err, "error getting Gateway: %v", err)
+		kubernetes.GatewayAcceptedConditionMustEqual(t, currentGW, metav1.ConditionTrue, v1beta1.GatewayReasonAccepted)
+		kubernetes.GatewayProgrammedConditionMustEqual(t, currentGW, metav1.ConditionFalse, v1beta1.GatewayReasonAddressNotUsable)
+		require.Len(t, currentGW.Status.Addresses, 1, "only one usable address was provided, so it should be the one reflected in status")
+		require.Equal(t, usableAddress, currentGW.Status.Addresses[0], "expected usable address to be assigned")
+
+		t.Logf("patching Gateway %s/%s to remove the unusable address", gwNN.Namespace, gwNN.Name)
+		updatedGW = currentGW.DeepCopy()
+		updatedGW.Spec.Addresses = filterAddr(currentGW.Spec.Addresses, unusableAddress)
+		err = s.Client.Patch(ctx, updatedGW, client.MergeFrom(currentGW))
+		require.NoError(t, err, "failed to patch Gateway: %v", err)
+		kubernetes.NamespacesMustBeReady(t, s.Client, s.TimeoutConfig, namespaces)
+		kubernetes.GatewayMustHaveLatestConditions(t, s.Client, s.TimeoutConfig, gwNN)
+
+		t.Logf("verifying that the Gateway %s/%s is accepted and programmed with the usable static address assigned", gwNN.Namespace, gwNN.Name)
+		err = s.Client.Get(ctx, gwNN, currentGW)
+		require.NoError(t, err, "error getting Gateway: %v", err)
+		kubernetes.GatewayAcceptedConditionMustEqual(t, currentGW, metav1.ConditionTrue, v1beta1.GatewayReasonAccepted)
+		kubernetes.GatewayProgrammedConditionMustEqual(t, currentGW, metav1.ConditionTrue, v1beta1.GatewayReasonProgrammed)
```

Comment:
For completeness, we should check all the positive-polarity conditions, hence we need to check `ResolvedRefs` as well.

Reply:
Added :+1: